src/components/support/controllers.tsx

```diff
@@ -71,10 +80,114 @@ interface ISignupForm extends ISignupFormValues {
   readonly values? : ISignupFormValues;
 }
 
+export interface IRequesterDetails {
+  readonly region?: string;
+  readonly acc_email?: string;
+  readonly roles?: ReadonlyArray<{
+    readonly orgGuid: string;
+    readonly orgName: string;
+    readonly roleType: string;
+  }>
+}
+
 const VALID_EMAIL = /[^.]@[^.]/;
 
 const TODAY_DATE = new Date();
 
+export async function fetchRequesterDetailsAndRoles(ctx: IContext): Promise<IRequesterDetails> {
+  // if user not logged in, bail and return empty object
+  const userLoggedIn =  ctx.session.passport?.user;
+  if (userLoggedIn === undefined ) return {};
+
+  const uaa = new UAAClient({
+    apiEndpoint: ctx.app.uaaAPI,
+    clientCredentials: {
+      clientID: ctx.app.oauthClientID,
+      clientSecret: ctx.app.oauthClientSecret,
+    },
+  });
+
+  const signingKeys = await uaa.getSigningKeys();
+  const token = new Token(ctx.session.passport.user, signingKeys);
+  const cf = new CloudFoundryClient({
+    accessToken: token.accessToken,
+    apiEndpoint: ctx.app.cloudFoundryAPI,
+    logger: ctx.app.logger,
+  });
+  const region = (ctx.app.location).toLowerCase();
+  // get user from UAA
+  const user = await uaa.getUser(token.userID);
+  // get their roles
+  const roles = await cf.userRoles(user!.id);
+  // fetch org details and roles the user has 
+  const userRoleTypeAndOrg = await Promise.all(
+    roles
+    .filter(r => r.relationships.organization.data)
+    .map(async r => ({
+      orgGuid: ((await cf.organization(r.relationships.organization.data.guid!)).metadata.guid),
+      orgName: ((await cf.organization(r.relationships.organization.data.guid!)).entity.name),
```

Comment:
I take it the cf client doesn't do any sort of caching. Could we re-use this lookup and so save us half the API requests?

I assume we're looking up the org's guid ... via its guid ... so we can be sure it exists? Otherwise we'd surely omit the request.

Reply:
cache no, but you're correct. we can just used the guid in the `data`  object.
amended